src/Hamiltonians/HubbardRealSpace.jl

```diff
@@ -53,58 +53,119 @@ end
 )
     return u * dot(occ_a, occ_b)
 end
-@inline function local_interaction(fs::CompositeFS, u, occs)
-    return _interactions(fs.components, u, occs)
+@inline local_interaction(::SingleComponentFockAddress, ::SingleComponentFockAddress,
+    ::Nothing,_,_) = 0
+"""
+    nearest_neighbor_interaction(onr1::SVector, onr2::SVector, w, geometry::CubicGrid, map1::ModeMap)
+
+Calculate the nearest neighbour interaction ``w \\sum_{⟨i,j⟩} n_{↑, i} n_{↓, j}`` for an occupation
+number representation `onr1` and `onr2` of the two single-component Fock states. For a
+multi-component Fock state, return the eigenvalue of
+
+```math
+\\frac{1}{2}\\sum_{⟨i,j⟩, σ, τ} w_{σ,τ} a^†_{σ,i}a^†_{τ,j}a_{τ,j}a_{σ,i} ,
+```
+
+where `w::SMatrix` is a symmetric matrix of interaction constants, `i, j` are the mode indices,
+and `σ`, `τ` are component indices.
+
+See also [`BoseFS`](@ref), [`FermiFS`](@ref), [`CompositeFS`](@ref).
+"""
+@inline function nearest_neighbor_interaction(onr1::SVector, onr2::SVector,
+    w, geometry::CubicGrid{D,S,B}, map1::ModeMap
+    ) where {D,S,B}
```

Comment:
```suggestion
@inline function nearest_neighbour_interaction(
    onr1::SVector, onr2::SVector, Δ, geometry::CubicGrid{D,S,B}, map1::ModeMap
) where {D,S,B}
```
It's better to break long lines like this. See [this style guide](https://github.com/JuliaDiff/BlueStyle?tab=readme-ov-file#method-definitions). Also please go through and make sure that your lines are not over 92 characters long. It makes the code easier to read.